enterprise/reporting/reporting.go

```diff
@@ -446,6 +451,27 @@ func (r *DefaultReporter) mainLoop(ctx context.Context, c types.SyncerConfig) {
 			if err != nil {
 				r.log.Errorf(`[ Reporting ]: Error deleting local reports from %s: %v`, ReportsTable, err)
 			}
+
+			vacuumStart := time.Now()
+			var sizeEstimate int64
+			if err := dbHandle.QueryRowContext(
+				ctx,
+				fmt.Sprintf(`SELECT pg_table_size(oid) from pg_class where relname='%s';`, ReportsTable),
+			).Scan(&sizeEstimate); err != nil {
+				r.log.Errorn(
+					`[ Reporting ]: Error getting table size estimate`,
+					logger.NewErrorField(err),
+				)
+			}
+			if sizeEstimate > config.GetInt64("Reporting.vacuumThresholdBytes", 5*bytesize.GB) {
+				if _, err := dbHandle.ExecContext(ctx, `vacuum full analyze reports;`); err != nil {
+					r.log.Errorn(
+						`[ Reporting ]: Error vacuuming reports table`,
+						logger.NewErrorField(err),
+					)
+				}
+			}
+			vacuumDuration.Since(vacuumStart)
```

Comment:
Should we send these only in case we are doing the vacuum? Basically from the above if condition?